### ISSUE TITLE
Wrong types have no effect on code execution is confusing

### ISSUE THREAD

**Issue by salvan13** [giltayar/proposal-types-as-comments#59]
If I understood correctly, with this approach we colud have wrong types but code will still work as always.
something like:
```
function equals(x: number, y: number): boolean {
    return x === y;
}

equals("a", "a");
// true
```
will still work, but it's very confusing especially for unexperienced people who approach the language, imho.

**Comment by benjamingr** [comment]
This is unprecedented in almost any language and most languages perform these checks in compile time and not runtime - the only difference is that here you get a choice of what type checker if any to use in your flow

**Comment by spenserblack** [comment]
Sorry, just want to clarify that when I said
>Just want to note that this isn't unprecedented in a language

I meant that it isn't unprecedented that a language *doesn't* check types at runtime instead of compile time. There are so many negatives in that last sentence that I'm confusing myself :laughing:, so let me state it in other words: there *is* a precedent that type hints are ignored at runtime. E.g., when Python introduced type hints, they *aren't* checked at runtime.

**Comment by wparad** [comment]
why would you have to denote the type checker? Wouldn't every tool to check types make the necessary adjustments to understand js types? Just as when new operands are introduced, no additional configuration is necessary, it shouldn't be necessary here either.

**Comment by ljharb** [comment]
@hax #36

**Comment by romulocintra** [comment]
Thanks, @salvan13 for opening this one,  As the readme says the proposal it's based on erasable types that the engine ignores, so tools are the ones that might catch those errors. Feel free to open it again if needed

**Comment by lillallol** [comment]
@romulocintra Why are you people acting like these:

> Can somebody just clearly, what is the expanded scope or problem statement? Is that the one that was flashed up in the slides or has there been
>
>RPR: The one on the slides and we will also consider runtime effects and simpler syntaxes.
>
>KG: And we'll consider changes that also require changes to TypeScript. That it's not just JS is necessarily going to try to move in the direction of the forks, but rather that we might try to find some common ground that both places can move towards and meet there.
>
>WH: Yeah, the clarification that was important to me was that much simpler syntaxes are also in scope and any compatibility with any fork is not a prerequisite.
>
>YSV: Is that clear for everyone?
>
>DRR: I think so.
>
>LCA: +1
>
>MWS: +1
>
>JHX: +1
>
>BN: Apollo supports stage 1.
>
>CDA: IBM supports stage 1.
>
>YSV: And the queue is empty, last call for any objections. [silence] Congratulations. Folks, you have stage 1.

were never stated?

**Comment by romulocintra** [comment]
> @romulocintra Why are you people acting like these:
> 
> > Can somebody just clearly, what is the expanded scope or problem statement? Is that the one that was flashed up in the slides or has there been
> > RPR: The one on the slides and we will also consider runtime effects and simpler syntaxes.
> > KG: And we'll consider changes that also require changes to TypeScript. That it's not just JS is necessarily going to try to move in the direction of the forks, but rather that we might try to find some common ground that both places can move towards and meet there.
> > WH: Yeah, the clarification that was important to me was that much simpler syntaxes are also in scope and any compatibility with any fork is not a prerequisite.
> > YSV: Is that clear for everyone?
> > DRR: I think so.
> > LCA: +1
> > MWS: +1
> > JHX: +1
> > BN: Apollo supports stage 1.
> > CDA: IBM supports stage 1.
> > YSV: And the queue is empty, last call for any objections. [silence] Congratulations. Folks, you have stage 1.
> 
> were never stated?

Hi, @lillallol  don't understand your question and how it relates to this issue, can you clarify?

**Comment by lillallol** [comment]
>  the proposal it's based on erasable types

That is not true. In fact the reason that the proposal got stage 1 is because it was agreed that it is not necessarily based on erasable types, and that we are open on runtime effects. Unfortunately people in this repository are acting like this was never stated.

**Comment by ljharb** [comment]
@lillallol that's not accurate. Stage 1 only means the committee wants to keep discussing the problem space; it simply does not imply that it is, or is not, based on erasable types or runtime effects or *literally any other semantics*, because that's not how the TC39 process works.

**Comment by lillallol** [comment]
> @lillallol that's not accurate.

Can you please be explicit on what `that` is referring to?

**Comment by ljharb** [comment]
I was referring to
> In fact the reason that the proposal got stage 1

**Comment by ctcpip** [comment]
@lillallol you referenced my +1 above.  it may be helpful to point out what I said earlier in that meeting:

> CDA: There's been some discussion about this proposal internally at IBM. We do share some of the concerns that WH is bringing up. There are varying opinions on the current content of the proposal, but we do all agree that it is worth exploring the problem area further, and that is specifically around expressing type information. We do think that's worthwhile, exploring that space. Again, some reservations about the proposal and the coupling to TypeScript. But if advancement to stage one is more about exploring the problem area, rather than signing off on the proposal as currently written, then we support it.

**Comment by lillallol** [comment]
@ljharb 

Sorry, but I am lost.

Have you read the tc39 meeting notes[[1](https://github.com/tc39/notes/blob/main/meetings/2022-03/mar-29.md#types-as-comments-for-stage-1)][[2](https://github.com/tc39/notes/blob/main/meetings/2022-03/mar-31.md#types-as-comments-continuation)]? Yes or no? It is obvious that if the champions were not about to come up with a new problem statement and extend it with: 

* runtime effects and simpler syntaxes
* any compatibility with any fork is not a prerequisite

the proposal would not have gotten stage1.

Also since `that` of 

> @lillallol that's not accurate.

is referring to

> In fact the reason that the proposal got stage 1

I can not understand why you mention:

> Stage 1 only means the committee wants to keep discussing the problem space; it simply does not imply that it is, or is not, based on erasable types or runtime effects or literally any other semantics, because that's not how the TC39 process works.

I can not find something I mentioned somewhere that explicitly or implicitly claims the opposite. In fact am the only one who seems to remind that to people.

You should tell that to people who claim that:

> the proposal it's based on erasable types [[3](https://github.com/tc39/proposal-type-annotations/issues/59#issuecomment-1210992725)]

> That has runtime semantics, which nothing in this proposal does [[4](https://github.com/tc39/proposal-type-annotations/issues/150#issuecomment-1145996425)]

**Comment by ljharb** [comment]
@lillallol i was present at the meeting, so I don't need to read the notes. The things you're listing are *not a problem statement*, and so they're not part of the proposal's stage 1 acceptance - they're just possible solutions to the problem statement. What I meant by "how the process works" was that a proposal being at stage 1 does not in fact lock in anything about the solution space.

**Comment by lillallol** [comment]
>The things you're listing are not a problem statement they're just possible solutions to the problem statement

I never set out to describe the problem statement in the first place. All I am saying is that the reason the context proposal has stage1 is because it was claimed that:

1. runtime effects and simpler syntaxes are in scope
2. any compatibility with any fork is not a prerequisite

and hence people were convinced to give stage 1.

Statements like these:

> the proposal it's based on erasable types

are in direct contradiction to the very reason this proposal has stage 1, but also to:

> stage 1 does not in fact lock in anything about the solution space.

However since you bring up the problem statement, this repo should have been updated with that, since April 2022, but people have not even bothered to add a link to the meeting notes.

Here is a conversation about that, which I find really strange:

>And furthermore, if we go forward with this are you going to expand on
this version of the problem statement in the The GitHub repo, which I don't think he
really talks about this really like at all?
>DRR: I don't want to spend too much time on this sort of meta question today. I
would really like to see if we can bring this to stage one and discuss, and then have a
better discussion in the future if that ends up being necessary. I do acknowledge that
may be a topic that is discussed in the future, though. So I appreciate you bringing it
up.
>EAO: So would you be updating the text the GitHub repo to reflect this, which is
currently not there at all? As a specific goal. Because right now, the goals that the
GitHub repos presenting about this are rather different.
>DRR: I don't think that we're aiming to work on processes.
RPR: Definitely I think in light of bringing this to plenary. And what we've heard about
the framing of the problem, the way that people in here would like to hear our goals
expressed. I think we've learned and will go back and update the repo certainly.

Isn't it common sense that a tc39 repo about a proposal has to be up to date with what is being decided about the proposal? Why EAO is asking about that? And why the answer is not a crystal clear and immediate YES, from DRR, and we instead get some generic answers to blatantly avoid answering the question?

The more I am getting my hands dirty with this proposal the more I feel I can not trust the people responsible for this repo.

**Comment by romulocintra** [comment]
Hi @lillallol , I believe this style of communication doesn't help for constructive and inclusive collaboration and this already happened in several threads you're involved in, I can list a few comments using an inappropriate tone referring to the work done by the authors, community and other users.

> The more I am getting my hands dirty with this proposal the more I feel I can not trust the people responsible for this repo.

The closed issue does not relate to the new thread created where I don't see any productive conclusion from it, other than the scrutiny of the plenary meeting notes with people that were at the meeting and people who weren't there.

The proposal's it's definitely Stage 1 and now we have room to investigate whatever is needed for this proposal to evolve,  and this might include,  drastic changes, new proposals, or even a demotion of the proposal can happen

We are going to talk about lots of different things during the  Stage 1 process and you are invited for that,  so I cannot understand your concern about having an opportunity to talk and study the problem space in a constructive manner. I foresee a long-stage process and all help is needed.  

Champions group and author's intention when the proposal was created was not to have runtime effects at all, but it's fair to understand others' motivation and be open for debate, historically those conversations are normally trigger for follow-on or new proposals. It's too early to guess what's going to happen but we should build it collectively. The closed issue might be reopened in the future if the direction of things and discussion requires it.
